### PR TITLE
Document multi-tenancy with operator

### PR DESCRIPTION
**Pull request by pavolloffay**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/tempo/setup/operator/multitenancy.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+menuTitle: Multi-tenancy
+weight: 250
+aliases:
+- /docs/tempo/operator/multitenancy
+---
+
+# Multi-tenancy
+
+Tempo is a multi-tenant distributed tracing backend. It supports multi-tenancy through the use of a header: `X-Scope-OrgID`.
+See [multi-tenancy docs]({{< relref "../../operations/multitenancy" >}}) for more details. This document outlines how to deploy and use multi-tenant Tempo with the Operator.
```
Comment (jdbaldry) [inline docs/sources/tempo/setup/operator/multitenancy.md:13]: Prefer "refer to X" over "see X".
 
 nit:  [Semantic line breaks](https://sembr.org/) are better for future line based diffing in the GitHub UI and result in the same HTML output. 
 
 ```suggestion
Tempo is a multi-tenant distributed tracing backend.
It supports multi-tenancy through the use of a header: `X-Scope-OrgID`.
For more information, refer to [Enable multitenancy]({{< relref "../../operations/multitenancy" >}}). 
This document outlines how to deploy and use multi-tenant Tempo with the Operator.
```

File: docs/sources/tempo/setup/operator/multitenancy.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+## Multitenancy without authentication
```
Comment (jdbaldry) [inline docs/sources/tempo/setup/operator/multitenancy.md:15]: Is it "multi-tenancy" or "multitenancy".

Perhaps @knylander-grafana can advise and we can lock this down in the Writers' Toolkit word list https://grafana.com/docs/writers-toolkit/write/style-guide/word-list/

File: docs/sources/tempo/setup/operator/multitenancy.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+The following example CR deploys multi-tenant Tempo instance. Note that Jaeger query is not tenant aware and therefore is not supported in this configuration.
```
Comment (jdbaldry) [inline docs/sources/tempo/setup/operator/multitenancy.md:17]: I think it makes sense to explain that abbreviation

```suggestion
The following example Kubernetes Custom Resource (CR) deploys a multi-tenant Tempo instance.
Note that Jaeger query is not tenant aware and therefore is not supported in this configuration.
```

File: docs/sources/tempo/setup/operator/multitenancy.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+  resources:
+    total:
+      limits:
+        memory: 2Gi
+        cpu: 2000m
+```
+
+## OIDC authentication with static RBAC
+
+On Kubernetes the multi-tenant Tempo instance uses OIDC authentication and static RABC authorization defined in the CR.
```
Comment (jdbaldry) [inline docs/sources/tempo/setup/operator/multitenancy.md:40]:
```suggestion
On Kubernetes, a multi-tenant Tempo instance uses OIDC authentication and static RBAC authorization defined in the CR.
```

File: docs/sources/tempo/setup/operator/multitenancy.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+        tenants:
+        - test-oidc
+```
+
+* The secret `oidc-test` defines fields `clientID`, `clientSecret` and `issuerCAPath`.
+* The RABC gives tenant `test-oidc` read and write access for traces. 
+
+## OpenShift
+
+On OpenShift the authentication and authorization does not require any 3rd party service dependencies.
```
Comment (jdbaldry) [inline docs/sources/tempo/setup/operator/multitenancy.md:97]:
```suggestion
On OpenShift, the authentication and authorization does not require any third party service dependencies.
```

File: docs/sources/tempo/setup/operator/multitenancy.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+The authentication uses OpenShift OAuth (the user is redirected to the OpenShift login page) and 
+authorization is handled through `SubjectAccessReview` (SAR).
```
Comment (jdbaldry) [inline docs/sources/tempo/setup/operator/multitenancy.md:99]: nit: 
[Semantic line breaks](https://sembr.org/) are better for future line based diffing in the GitHub UI and result in the same HTML output.
```suggestion
The authentication uses OpenShift OAuth (the user is redirected to the OpenShift login page) and authorization is handled through `SubjectAccessReview` (SAR).
```

File: docs/sources/tempo/setup/operator/multitenancy.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+        tenantId: "1610b0c3-c509-4592-a256-a1871353dbfb"
+  template:
+    gateway:
+      enabled: true
+    queryFrontend:
+      jaegerQuery:
+        enabled: true
+```
+
+User is responsible for creating `ClusterRole` and `ClusterRoleBinding` for reading and writing the data.
```
Comment (jdbaldry) [inline docs/sources/tempo/setup/operator/multitenancy.md:131]: Who is the user here? Can we explain this a little more?

File: docs/sources/tempo/setup/operator/multitenancy.md (hunk 2 above)
Comment (pavolloffay) [reply docs/sources/tempo/setup/operator/multitenancy.md:15]: multi-tenancy is used in other pages

Comment (jdbaldry) [reply docs/sources/tempo/setup/operator/multitenancy.md:15]:
```suggestion
## Multi-tenancy without authentication
```

**Comment by jdbaldry** [comment]
Thanks for all the improvements! I think it would be best to get @knylander-grafana to also take a look.

File: docs/sources/tempo/setup/operator/multitenancy.md [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+# Enable multi-tenancy
+
+Tempo is a multi-tenant distributed tracing backend. It supports multi-tenancy through the use of a header: `X-Scope-OrgID`.
+Refer to [multi-tenancy docs]({{< relref "../../operations/multitenancy" >}}) for more details. 
+This document outlines how to deploy and use multi-tenant Tempo with the Operator.
+
+## Multi-tenancy without authentication
+
+The following Kubernetes Custom Resource (CR) deploys a multi-tenant Tempo instance.
+Note that Jaeger query is not tenant aware and therefore is not supported in this configuration.
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/operator/multitenancy.md:19]: Changing the note to use the admonition shortcode. 
```suggestion
{{% admonition type="note" %}}
Jaeger query is not tenant aware and therefore is not supported in this configuration.
{{% /admonition %}}
```

File: docs/sources/tempo/setup/operator/multitenancy.md [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+        memory: 2Gi
+        cpu: 2000m
+```
+
+## OIDC authentication with static RBAC
+
+On Kubernetes, a multi-tenant Tempo instance uses OIDC authentication and static RBAC authorization defined in the CR.
+The instance should be accessed through service `tempo-simplest-gateway` which handles authentication and authorization.
+The service exposes Jaeger query API and OpenTelemetry gRPC (OTLP) for trace ingestion.
+The Jaeger UI can be accessed at `http://<exposed gateway service>:8080/api/traces/v1/<tenant-name>/search`
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/operator/multitenancy.md:45]:
```suggestion
The Jaeger UI can be accessed at `http://<exposed gateway service>:8080/api/traces/v1/<tenant-name>/search`.
```

File: docs/sources/tempo/setup/operator/multitenancy.md [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+        - read
+        - write
+        resources:
+        - traces
+        tenants:
+        - test-oidc
+```
+
+* The secret `oidc-test` defines fields `clientID`, `clientSecret` and `issuerCAPath`.
+* The RABC gives tenant `test-oidc` read and write access for traces.
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/operator/multitenancy.md:95]: Should this be RBAC?

```suggestion
* The RBAC gives tenant `test-oidc` read and write access for traces. 
```

File: docs/sources/tempo/setup/operator/multitenancy.md [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+        tenants:
+        - test-oidc
+```
+
+* The secret `oidc-test` defines fields `clientID`, `clientSecret` and `issuerCAPath`.
+* The RABC gives tenant `test-oidc` read and write access for traces. 
+
+## OpenShift
+
+On OpenShift, the authentication and authorization does not require any third party service dependencies.
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/operator/multitenancy.md:99]:
```suggestion
On OpenShift, the authentication and authorization does not require any third-party service dependencies.
```

File: docs/sources/tempo/setup/operator/multitenancy.md [hunk 12]
```diff
@@ -0,0 +1,227 @@
...
+The authentication uses OpenShift OAuth (the user is redirected to the OpenShift login page) and authorization is handled through `SubjectAccessReview` (SAR).
+
+The instance should be accessed through service `tempo-simplest-gateway` which handles authentication and authorization.
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/operator/multitenancy.md:102]:
```suggestion
The instance should be accessed through service `tempo-simplest-gateway`, which handles authentication and authorization.
```

File: docs/sources/tempo/setup/operator/multitenancy.md [hunk 13]
```diff
@@ -0,0 +1,227 @@
...
+The service exposes Jaeger query API and OpenTelemetry gRPC (OTLP) for trace ingestion.
+The Jaeger UI can be accessed at `http://<exposed gateway service>:8080/api/traces/v1/<tenant-name>/search`
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/operator/multitenancy.md:104]:
```suggestion
The instance should be accessed through service `tempo-simplest-gateway`, which handles authentication and authorization.
The service exposes Jaeger query API and OpenTelemetry gRPC (OTLP) for trace ingestion.
The Jaeger UI can be accessed at `http://<exposed gateway service>:8080/api/traces/v1/<tenant-name>/search`.
```

**Review by knylander-grafana — APPROVED** [review]
Thank you for adding docs. I'm approving with some minor grammatical suggestions.

**Comment by pavolloffay** [comment]
@knylander-grafana PR updated. Thanks for the review

**Comment by pavolloffay** [comment]
This can be backported to 2.2 as  well

**Comment by knylander-grafana** [comment]
> @knylander-grafana PR updated. Thanks for the review
Thank you for updating docs! I can merge it now for you, if you like.

**Pull request merged**